4.component.md

```diff
@@ -74,573 +48,72 @@ The Parameters section defines all of the configurable parameters for this compo
       Sample component schematic that describes the administrative interface for our Twitter bot.
 spec:
-  workloadType: Server
-  osType: linux
-  parameters:
-  - name: username
-    description: Basic auth username for accessing the administrative interface
-    type: string
-    required: true
-  - name: password
-    description: Basic auth password for accessing the administrative interface
-    type: string
-    required: true
-  - name: backend-address
-    description: Host name or IP of the backend
-    type: string
-    required: true
-  containers:
-  - name: my-twitter-bot-frontend
-    image:
-      name: example/my-twitter-bot-frontend:1.0.0
-      digest: sha256:6c3c624b58dbbcd3c0dd82b4c53f04194d1247c6eebdaab7c610cf7d66709b3b
-    resources:
-      cpu:
-        required: 1.0
-      memory:
-        required: 100MB
-    ports:
-    - name: http
-      value: 8080
-    env:
-    - name: USERNAME
-      fromParam: 'username'
-    - name: PASSWORD
-      fromParam: 'password'
-    - name: BACKEND_ADDRESS
-      fromParam: 'backend-address'
-    livenessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
-    readinessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
-```
-
-Our second component implements both the backend for our Twitter bot's
-administrative interface and a background worker process that will Tweet
-end-user-defined content at end-user-defined intervals (defined through the
-administrative interface). This component also persists its end-user-defined
-configuration to disk as a JSON file.
-
-This component's design prevents it from scaling horizontally so we should only
-have at most _one_ instance of it running. It also has an endpoint for an administrative interface. As such, this component is best described
-as a `SingletonServer`. Connection details for the Twitter API are
-configurable. The component schematic also describes the file system location
-where the component persists end-user-defined configuration.
-
-```yaml
-apiVersion: core.oam.dev/v1alpha1
-kind: ComponentSchematic
-metadata:
-  name: admin-backend
-  annotations:
-    version: v1.0.0
-    description: >
-      Sample component schematic that describes the backend for our Twitter bot.
-spec:
-  workloadType: core.oam.dev/v1.SingletonServer
-  osType: linux
-  parameters:
-  - name: twitter-consumer-key
-    description: Twitter API consumer key
-    type: string
-    required: true
-  - name: twitter-consumer-secret
-    description: Twitter API consumer secret
-    type: string
-    required: true
-  - name: twitter-access-token
-    description: Twitter API access token
-    type: string
+  workload:
+    apiVersion: core.oam.dev/v1alpha2
+    kind: ContainerizedWorkload
+    metadata:
+      name: sample-workload
+    spec:
+      osType: linux
+      containers:
+      - name: my-cool-workload
+        image:
+          name: example/very-cool-workload:0.1.2
+          digest: sha256:verytrustworthyhash
```

Comment:
```suggestion
        image: example/very-cool-workload:0.1.2@sha256:verytrustworthyhash
```
Kubernetes uses [docker syntax](https://docs.docker.com/engine/reference/commandline/pull/#pull-an-image-by-digest-immutable-identifier#pull-an-image-by-digest-immutable-identifier) for pulling images by digest.